code/API/FreshdeskAPI.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+
+class FreshdeskAPI extends \Object
```

Comment:
It feels somehow unfinished that you are injecting the client, but then creating `FreshdeskAPI` dynamically later. It seems like only one copy of `FreshdeskAPI` would be necessary at any given time, which means it would be best to inject it into the controllers.

It would be worth trying if you can merge FreshdeskAPI and FreshdeskClientFactory into one object. It doesn't need to be a factory - I believe you should be able to instantiate the Guzzle Client in the constructor and assign against local variable.

Reply:
ack